### DISCUSSION TITLE
Quay doing a 301 Redirect

### DISCUSSION THREAD

**Discussion by sabman3** [quay/quay#3016] [Q&A]
I went through and followed the instructions to get Quay working with podman (https://github.com/quay/quay/blob/master/docs/quick-local-deployment.md) and I still cannot get it working. When I go to http://quay:8080/v2 or use podman login, it redirects me to https://quay/v2. I've tried turning off HSTs in the browser, but that did not work. Looking at a trace it appears to be doing a 301 redirect.

# podman login --tls-verify=false quay:8080
Username: quay
Password: 
Error: authenticating creds for "quay:8080": pinging container registry quay:8080: Get "https://quay/v2/": dial tcp 127.0.0.1:443: connect: connection refused

**Comment by ibazulic** [comment]
Hi There.

To use HTTP only, you need to set 

```
PREFERRED_URL_SCHEME: http
```

in your Quay's `config.yaml` file. Additionally, Quay is not meant to be used with short hostnames, only FQDN is supported. And it needs to be set in

```
SERVER_HOSTNAME: quay.example.com:8080
```

HTH!

**Reply by sabman3** [reply]
I change to a the FQDN and made sure the URL SCHEME is setup correctly. Still getting a redirect. I'm not that much of a gunicorn or nginx expert to dive deeper. Also nginx access log is empty, which seem weird. (attached config.yaml)

[config.txt](https://github.com/user-attachments/files/16169457/config.txt)

podman login --tls-verify=false quay.example.com:8080
Username: quay
Password: 
Error: authenticating creds for "quay.example.com:8080": pinging container registry quay.example.com:8080: Get "https://quay.example.com/v2/": dial tcp 127.0.0.1:443: connect: connection refused

telnet quay.example.com 8080
Trying 127.0.0.1...
Connected to quay.srp.gov.
Escape character is '^]'.
GET /v2 HTTP/1.0

HTTP/1.1 301 Moved Permanently
Server: nginx/1.20.1
Date: Wed, 10 Jul 2024 22:06:39 GMT
Content-Type: text/html
Content-Length: 169
Connection: close
Location: https://_/v2

<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx/1.20.1</center>
</body>
</html>
Connection closed by foreign host.
